src/replication/coordinator.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include "replication/coordinator.hpp"
+
+namespace memgraph::replication {
+
+#ifdef MG_ENTERPRISE
+utils::BasicResult<RegisterReplicaError, ReplicationClient *> Coordinator::RegisterReplica(
+    const ReplicationClientConfig &config) {
```

Comment:
The coordinator should use its own config.

Reply:
Out of scope atm, here we don't introduce any kind of networking.